app/actions/views/channel.js

```diff
@@ -123,8 +124,12 @@ export function selectInitialChannel(teamId) {
 export function handleSelectChannel(channelId) {
     return async (dispatch, getState) => {
         const currentTeamId = getState().entities.teams.currentId;
+
         await updateStorage(currentTeamId, {currentChannelId: channelId});
         await selectChannel(channelId)(dispatch, getState);
+        setTimeout(async () => {
```

Comment:
For no other reason than that's what seemed to be the happy medium between waiting to long for the transition to occur and the transition being choppy.